apps/api-extractor/src/generators/ExcerptBuilder.ts

```diff
@@ -113,8 +118,12 @@ export class ExcerptBuilder {
 
     if (span.prefix) {
       if (span.kind === ts.SyntaxKind.Identifier) {
+        const name: ts.Identifier = span.node as ts.Identifier;
+        const canonicalReference: DeclarationReference | undefined = isDeclarationName(name)
+          ? undefined
+          : state.referenceGenerator.getDeclarationReferenceForIdentifier(name);
         ExcerptBuilder._appendToken(excerptTokens, ExcerptTokenKind.Reference,
```

Comment:
BTW for an input like this:
```ts
export function exampleFunction(x: number): number {
  return x;
}
```

...it was an accident that identifiers like `exampleFunction` and `x` are being represented as `ExcerptTokenKind.Reference`.  The initial implementation merely sought to demonstrate the `Content`/`Reference` distinction for consumers of the API, but without actually emitting any link targets.  It simply assigned `ExcerptTokenKind.Reference`  whenever `span.kind === ts.SyntaxKind.Identifier`, without considering the kind of identifier.

Instead of this:
```js
          "excerptTokens": [
            {
              "kind": "Content",
              "text": "export declare function "
            },
            {
              "kind": "Reference",
              "text": "exampleFunction"
            },
            {
              "kind": "Content",
              "text": "("
            },
            {
              "kind": "Reference",
              "text": "x"
            },
            {
              "kind": "Content",
              "text": ": "
            },
            {
              "kind": "Content",
              "text": "number"
            },
            {
              "kind": "Content",
              "text": "): "
            },
            {
              "kind": "Content",
              "text": "number"
            },
            {
              "kind": "Content",
              "text": ";"
            }
          ],
```
...the `"kind"` should really be `"Content"` everywhere.  We should only use `"Reference"` when there we have an actual `"canonicalReference"` link target.

(BTW if it turns out to be useful to distinguish `"exampleFunction"`  versus `"): ", then I'd suggest to introduce a new kind like `ExcerptTokenKind.SimpleIdentifier` for identifiers that lack a `canonicalReference`.)  We could also maybe convert `IExcerptToken` into a discriminated union based on `ExcerptTokenKind`, to make that more clear.

Reply:
The problem I see is that the `ExcerptBuilder` doesn't know anything about the token in question. If we remove the `canonicalReference` for `exampleFunction` here, then we would have nothing to link to if we built an excerpt for the type `typeof exampleFunction`. It may be extraneous, but I don't think we should remove it.

Reply:
Take a look at PR https://github.com/rbuckton/web-build-tools/pull/2 (into your PR branch).

I changed this:

[src/generators/ExcerptBuilder.ts ](https://github.com/rbuckton/web-build-tools/pull/2/commits/a2e45837f7971fef351acf943afebab77b48b3f6)
```ts
      if (span.kind === ts.SyntaxKind.Identifier) {
        const name: ts.Identifier = span.node as ts.Identifier;
        const canonicalReference: DeclarationReference | undefined = isDeclarationName(name)
          ? undefined
          : state.referenceGenerator.getDeclarationReferenceForIdentifier(name);
        ExcerptBuilder._appendToken(excerptTokens, ExcerptTokenKind.Reference,
          span.prefix, state, canonicalReference);
      } else {
        ExcerptBuilder._appendToken(excerptTokens, ExcerptTokenKind.Content,
          span.prefix, state);
      }
```
...to this:
```ts
      let canonicalReference: DeclarationReference | undefined = undefined;

      if (span.kind === ts.SyntaxKind.Identifier) {
        const name: ts.Identifier = span.node as ts.Identifier;
        if (!isDeclarationName(name)) {
          canonicalReference = state.referenceGenerator.getDeclarationReferenceForIdentifier(name);
        }
      }

      if (canonicalReference) {
        ExcerptBuilder._appendToken(excerptTokens, ExcerptTokenKind.Reference,
          span.prefix, state, canonicalReference);
      } else {
        ExcerptBuilder._appendToken(excerptTokens, ExcerptTokenKind.Content,
          span.prefix, state);
      }
```

The [test-outputs/typeOf2](https://github.com/rbuckton/web-build-tools/blob/585184975a21435ca9a88410f7b03b05e64ab097/build-tests/api-extractor-scenarios/etc/test-outputs/typeOf2/api-extractor-scenarios.api.json#L58) still correctly links `typeof f`:

```ts
            {
              "kind": "Content",
              "text": "typeof "
            },
            {
              "kind": "Reference",
              "text": "f",
              "canonicalReference": "api-extractor-scenarios!f:function"
            },
```